ksqldb-engine/src/main/java/io/confluent/ksql/engine/EngineExecutor.java

```diff
@@ -187,16 +188,16 @@ PullQueryResult executeTablePullQuery(
       final KsqlConfig ksqlConfig = sessionConfig.getConfig(false);
       final LogicalPlanNode logicalPlan = buildAndValidateLogicalPlan(
           statement, analysis, ksqlConfig, queryPlannerOptions, false);
-      final PullPhysicalPlan physicalPlan = buildPullPhysicalPlan(
+
+      plan = buildPullPhysicalPlan(
           logicalPlan,
           analysis
       );
-      sourceType = physicalPlan.getSourceType();
-      planType = physicalPlan.getPlanType();
```

Comment:
These are either both or neither present, so I just set the whole `plan` instead.